app.js

```diff
@@ -79,7 +95,58 @@ function roomData(req, res, next) {
       returnDataOr404(res, data);
     })
     .catch((err) => {
-      console.log(err)
+      console.log(err);
       next(err);
     });
+}
+
+// Login function, POST request with email and password
+function auth(req, res, next) {
+  const email = req.body.email;
+  const username = req.body.email.split('@')[0]
+  const password = req.body.password;
+  console.log(req.body)
+  if (email && password) {
+    db.oneOrNone(`SELECT pass FROM accounts WHERE email = $1`, [
+      email,
+    ]) // Request the hashed password from the db
+      .then((data) => {
+        if (data) {
+          console.log(data.pass)
+
+          if (bcrypt.compareSync(password, data.pass)) { // compare the hashed password with the login password
+            const accessToken = jwt.sign({ username: username }, accessTokenSecret) // create a signed jwt
```

Comment:
What exactly is a jwt? Also, you said the default username is the stuff before the @ in the email, are we just grabbing that or are we accepting the full emails as well?

Reply:
JWTs are a way to authenticate a user, basically make sure on the front end that a user is good to go and logged in.

>JSON Web Token is an Internet standard for creating data with optional signature and/or optional encryption whose payload holds JSON that asserts some number of claims. The tokens are signed either using a private secret or a public/private key.